### PR TITLE
Always wipe DB state after benchmark

### PR DESCRIPTION
**Pull request by ggwpez**
Currently the DB state is not wiped in case that the benchmark returned an error.  
This is only a problem when the benchmark is invoked manually. Fixed it by using the new `defer!` :raised_hands:   

Thanks to @AurevoirXavier for bringing attention to this. There is another oddity with the block number initialization which will be fixed in a new MR.

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]
nice, very clever

**Review by AurevoirXavier — APPROVED** [review]

File: frame/benchmarking/src/tests.rs [hunk 1]
```diff
@@ -125,6 +125,8 @@ fn new_test_ext() -> sp_io::TestExternalities {
 	GenesisConfig::default().build_storage().unwrap().into()
 }
 
+// NOTE: This attribute is only needed for the `modify_in_` functions.
+#[allow(unreachable_code)]
```
Comment (bkchr) [inline frame/benchmarking/src/tests.rs:129]: Is this because of code that is being generated by the macro?

**Review by bkchr — APPROVED** [review]

File: frame/benchmarking/src/tests.rs (hunk 1 above)
Comment (ggwpez) [reply frame/benchmarking/src/tests.rs:129]: Yep. It is placing code after the `return Err(…)`.

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**